tests/integration/Paginator/Adapter/QueryBuilderCest.php

```diff
@@ -136,33 +138,64 @@ public function testIssue12111WithoutGroup(IntegrationTester $I)
      */
     public function testIssue12957(IntegrationTester $I)
     {
-        $I->skipTest('TODO: This needs to be checked for logic');
-        $this->specify(
-            "Query builder paginator doesn't work correctly with a different db service",
-            function () {
-                $modelsManager = $this->setUpModelsManager();
-                $di = $modelsManager->getDI();
-                $di->set(
-                    'dbTwo',
-                    $di->get('db')
-                );
-                $builder = $modelsManager->createBuilder()
-                    ->columns("COUNT(*) as robos_count")
-                    ->from(['Robos' => Robos::class])
-                    ->groupBy('type')
-                    ->having('MAX(Robos.year) > 1970');
-
-                $paginate = (new QueryBuilder(
-                    [
-                        "builder" => $builder,
-                        "limit"   => 1,
-                        "page"    => 2
-                    ]
-                ))->paginate();
+        $this->setDiMysql();
+        $modelsManager = $this->getService('modelsManager');
 
-                expect($paginate->last)->equals(2);
-                expect($paginate->total_items)->equals(2);
-            }
+        $di = $this->getDI();
```

Comment:
`$di = $this->getDI()` => `$this->container`

So you can do `$this->container->set('dbTwo', $this->container->get('db'));`